pinot-query-planner/src/main/java/org/apache/pinot/query/planner/serde/RexExpressionVisitor.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.planner.serde;
+
+import com.google.protobuf.ByteString;
+import java.io.Serializable;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.SerializationUtils;
+import org.apache.pinot.common.proto.Expressions;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.query.planner.logical.RexExpression;
+import org.apache.pinot.spi.utils.ByteArray;
+
+
+/**
+ * Serialize RexExpressions to the Protobuf equivalent classes defined expressions.proto.
+ */
+public class RexExpressionVisitor {
```

Comment:
The previous pattern worked because it could use recursion. This pattern needs 4 methods to handle 4 implementations of RexExpression. Another 4 methods to deserialize. Splitting them up is more readable for me. I've changed the names to `RexExpressionToProtoExpression` and `ProtoExpressionToRexExpression`